Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py

```diff
@@ -1571,54 +1579,60 @@ def long_running_loop():
             else:
-                port = int(port_mapping)
-        else:
-            raise ValueError('No port mapping was provided')
-        Thread(target=channel_mirror_loop, daemon=True).start()
-        demisto.info('Started channel mirror loop thread')
-
-        ssl_args = dict()
-
-        if certificate and private_key:
-            certificate_file = NamedTemporaryFile(delete=False)
-            certificate_path = certificate_file.name
-            certificate_file.write(bytes(certificate, 'utf-8'))
-            certificate_file.close()
-            ssl_args['certfile'] = certificate_path
-
-            private_key_file = NamedTemporaryFile(delete=False)
-            private_key_path = private_key_file.name
-            private_key_file.write(bytes(private_key, 'utf-8'))
-            private_key_file.close()
-            ssl_args['keyfile'] = private_key_path
-
-            demisto.info('Starting HTTPS Server')
-        else:
-            demisto.info('Starting HTTP Server')
+                raise ValueError('No port mapping was provided')
+            Thread(target=channel_mirror_loop, daemon=True).start()
+            demisto.info('Started channel mirror loop thread')
+
+            ssl_args = dict()
+
+            if certificate and private_key:
+                certificate_file = NamedTemporaryFile(delete=False)
+                certificate_path = certificate_file.name
+                certificate_file.write(bytes(certificate, 'utf-8'))
+                certificate_file.close()
+                ssl_args['certfile'] = certificate_path
+
+                private_key_file = NamedTemporaryFile(delete=False)
+                private_key_path = private_key_file.name
+                private_key_file.write(bytes(private_key, 'utf-8'))
+                private_key_file.close()
+                ssl_args['keyfile'] = private_key_path
+
+                demisto.info('Starting HTTPS Server')
+            else:
+                demisto.info('Starting HTTP Server')
 
-        server = WSGIServer(('0.0.0.0', port), APP, **ssl_args)
-        server.serve_forever()
-    except Exception as e:
-        if certificate_path:
-            os.unlink(certificate_path)
-        if private_key_path:
-            os.unlink(private_key_path)
-        demisto.error(f'An error occurred in long running loop: {str(e)}')
-        raise ValueError(str(e))
+            server = WSGIServer(('0.0.0.0', port), APP, **ssl_args)
+            server.serve_forever()
+        except Exception as e:
+            error_message = str(e)
+        finally:
+            if certificate_path:
+                os.unlink(certificate_path)
+            if private_key_path:
+                os.unlink(private_key_path)
+            if error_message:
+                demisto.error(f'An error occurred in long running loop: {error_message}')
+                demisto.updateModuleHealth(f'An error occurred: {error_message}')
+            time.sleep(5)
```

Comment:
do we want to loop again after we unlinked all the certificates?

Reply:
they will be created again in the next iteration, so i think its ok

Reply:
The files may be already exist no?

Reply:
not sure i follow, can you give an example ?

Reply:
never mind, it's a temporary file